File: app/src/main/java/com/ywwynm/everythingdone/adapters/ImageViewerPagerAdapter.java
package com.ywwynm.everythingdone.adapters;

import android.support.v4.view.PagerAdapter;
import android.support.v4.view.ViewPager;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

/**
 * Created by ywwynm on 2015/10/11.
 * A subclass of {@link PagerAdapter} for {@link android.support.v4.view.ViewPager}
 * in {@link com.ywwynm.everythingdone.activities.ImageViewerActivity}
 */
public class ImageViewerPagerAdapter extends PagerAdapter {

    public static final String TAG = "ImageViewerPagerAdapter";

    private List<View> mTabs;

    public ImageViewerPagerAdapter(List<View> tabs) {
        mTabs = tabs;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        final View view = mTabs.get(position);
        container.addView(view, 0);
        return view;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView(mTabs.get(position));
    }

    @Override
    public int getCount() {
        return mTabs.size();
    }

    @Override
    public int getItemPosition(Object object) {
        return POSITION_NONE;
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }

    public void removeTab(ViewPager viewPager, int index) {
        viewPager.setAdapter(null);
        mTabs.remove(index);
        viewPager.setAdapter(this);
        viewPager.setCurrentItem(index);
    }
}

File: app/src/main/java/com/ywwynm/everythingdone/adapters/ImageViewerPagerAdapter.kt
package com.ywwynm.everythingdone.adapters

import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager
import android.view.View
import android.view.ViewGroup

/**
 * Created by ywwynm on 2015/10/11.
 * Translated to Kotlin by ywwynm and Claude Opus 4.7 on 2026/5/20.
 * A subclass of [PagerAdapter] for [androidx.viewpager.widget.ViewPager]
 * in [com.ywwynm.everythingdone.activities.ImageViewerActivity]
 */
open class ImageViewerPagerAdapter(tabs: List<View?>?) : PagerAdapter() {

    private var mTabs: MutableList<View?>? = tabs as MutableList<View?>?

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view: View = mTabs!![position]!!
        container.addView(view, 0)
        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(mTabs!![position])
    }

    override fun getCount(): Int = mTabs!!.size

    override fun getItemPosition(`object`: Any): Int = POSITION_NONE

    override fun isViewFromObject(view: View, `object`: Any): Boolean = view === `object`

    open fun removeTab(viewPager: ViewPager?, index: Int) {
        viewPager!!.adapter = null
        mTabs!!.removeAt(index)
        viewPager.adapter = this
        viewPager.currentItem = index
    }

    companion object {
        const val TAG: String = "ImageViewerPagerAdapter"
    }
}
